src/uthash.h

```diff
@@ -699,28 +699,29 @@ typedef unsigned char uint8_t;
-            case 1:                                                                                          \
-                _hj_i += _hj_key[0]; /* FALLTHROUGH */                                                       \
-            default:;                                                                                        \
+                hashv += ((unsigned)_hj_key[10] << 24);                                                      \
+                H5_ATTR_FALLTHROUGH /* FALLTHROUGH */                                                        \
+                    case 10 : hashv += ((unsigned)_hj_key[9] << 16);                                         \
+                H5_ATTR_FALLTHROUGH /* FALLTHROUGH */                                                        \
+                    case 9 : hashv += ((unsigned)_hj_key[8] << 8);                                           \
+                H5_ATTR_FALLTHROUGH /* FALLTHROUGH */                                                        \
+                    case 8 : _hj_j += ((unsigned)_hj_key[7] << 24);                                          \
+                H5_ATTR_FALLTHROUGH /* FALLTHROUGH */                                                        \
+                    case 7 : _hj_j += ((unsigned)_hj_key[6] << 16);                                          \
+                H5_ATTR_FALLTHROUGH /* FALLTHROUGH */                                                        \
+                    case 6 : _hj_j += ((unsigned)_hj_key[5] << 8);                                           \
+                H5_ATTR_FALLTHROUGH /* FALLTHROUGH */                                                        \
+                    case 5 : _hj_j += _hj_key[4];                                                            \
+                H5_ATTR_FALLTHROUGH /* FALLTHROUGH */                                                        \
+                    case 4 : _hj_i += ((unsigned)_hj_key[3] << 24);                                          \
+                H5_ATTR_FALLTHROUGH /* FALLTHROUGH */                                                        \
+                    case 3 : _hj_i += ((unsigned)_hj_key[2] << 16);                                          \
+                H5_ATTR_FALLTHROUGH /* FALLTHROUGH */                                                        \
+                    case 2 : _hj_i += ((unsigned)_hj_key[1] << 8);                                           \
+                H5_ATTR_FALLTHROUGH /* FALLTHROUGH */                                                        \
+                    case 1 : _hj_i += _hj_key[0];                                                            \
+                H5_ATTR_FALLTHROUGH /* FALLTHROUGH */                                                        \
+                    default:;                                                                                \
```

Comment:
It turns out we do need the attribute keyword. clang doesn't respect the comments.